File: python/Data_structures/stack.py
class Stack:

    def __init__(self):
        self._data = []
    
    def push(self, x):
        self._data.append(x)
    
    def pop(self):
        x = self._data[len(self._data) - 1]
        del self._data[len(self._data) - 1]
        return x
    
    def peek(self):
        return self._data[len(self._data) - 1]
    
    def empty(self):
        self._data = []
    

def test():
    stack = Stack()
    stack.push(1)
    stack.push(2)
    stack.push(3)
    print(stack.pop())
    print(stack.pop())
    stack.empty()
    stack.push(4)
    stack.push(5)
    print(stack.peek())
    stack.push(6)
    print(stack.pop())
    print(stack.pop())
    print(stack.pop())


if __name__ == '__main__':
    test()

File: php/data-structure/Stack.php
<?php
/**
 * Data Structure: Stack
 * Property: Last-In-First-Out
 * Methods: Size, Push, Pop, Top, Traverse
 */

class Stack {

    // Member Variables
    private $size = 0;
    private $list = [];

    function __construct($list = [])
    {
        if (is_array($list) && count($list) > 0) {
            $this->list = $list;
            $this->size = count($list);
        }
    }

    // Returns Size of The Stack
    public function size()
    {
        return $this->size;
    }

    // Pushes new Element to Stack
    public function push($value = null)
    {
        if ($value != null) {
            $this->list[] = $value;
            $this->size++;
            return 1;
        }
        return 0;
    }

    // Pops last Element from Stack
    public function pop()
    {
        if ($this->size > 0) {
            $this->size--;
            return array_pop($this->list);
        }
        return null;
    }

    // Return Top Element of Stack
    public function top()
    {
        if ($this->size > 0) {
            return $this->list[$this->size - 1];
        }
        return null;
    }

    // Traverse all Elements in Stack
    public function traverse(callable $callback)
    {
        if ($this->size > 0) {
            return array_map($callback, array_reverse($this->list));
        }
        return [];
    }
}

/**
 * Test Suite
 */
$test_stack = new Stack();

$test_stack->push(5);
$test_stack->push("Hello");
$test_stack->push("World");
$test_stack->push(8);

echo $test_stack->pop();
echo $test_stack->top();

$test_stack->traverse(function($el) {
    echo $el;
});

?>
